pysatMadrigal/instruments/methods/madrigal.py

```diff
@@ -58,107 +54,217 @@ def load(fnames, tag=None, inst_id=None, xarray_coords=[]):
+                data = data.assign(time=pds.Series(time, index=data.index))
+
+                # Cycle through each of the coordinate dimensions
+                xdatasets = list()
+                for xcoords in coord_order:
+                    if not np.all([xkey.lower() in data.columns
+                                   for xkey in xcoords]):
+                        raise ValueError(''.join(['unknown coordinate key in ',
+                                                  '[{:}], use'.format(xcoords),
+                                                  ' only: {:}'.format(
+                                                      data.columns)]))
+                    if not np.all([xkey.lower() in data.columns
+                                   for xkey in xarray_coords[xcoords]]):
+                        data_mask = [xkey.lower() in data.columns
+                                     for xkey in xarray_coords[xcoords]]
+                        if np.all(~np.array(data_mask)):
+                            raise ValueError(''.join(['all provided data ',
+                                                      'variables [',
+                                                      '{:}] are '.format(
+                                                          xarray_coords[
+                                                              xcoords]),
+                                                      'unknown, use only: ',
+                                                      '{:}'.format(
+                                                          data.columns)]))
+                        else:
+                            logger.warning(''.join(['unknown data variable in',
+                                                    ' [{:}], use only'.format(
+                                                        xarray_coords[xcoords]),
+                                                    ': {:}'.format(
+                                                        data.columns)]))
+
+                            # Remove the coordinates that aren't present
+                            temp = np.array(xarray_coords[xcoords])[data_mask]
+                            xarray_coords[xcoords] = list(temp)
+
+                    # Select the desired data values
+                    sel_data = data[list(xcoords) + xarray_coords[xcoords]]
+
+                    # Remove duplicates before indexing, to ensure data with
+                    # the same values at different locations are kept
+                    sel_data = sel_data.drop_duplicates()
+
+                    # Set the indices
+                    sel_data = sel_data.set_index(list(xcoords))
+
+                    # Recast as an xarray
+                    xdatasets.append(sel_data.to_xarray())
+
+                # Merge all of the datasets
+                for i in np.arange(1, len(xdatasets)):
+                    xdatasets[0] = xdatasets[0].merge(xdatasets[i])
```

Comment:
```suggestion
                # Merge all of the datasets
                if len(xdatasets) > 1:
                    xdatasets = xr.merge(xdatasets, dim='time')
                else:
                    xdatasets = xdatasets[0]
```

I wasn't able to get hdf5 support going so I haven't been able to directly test this.